ai_agent_tutorials/ai_health-fitness_agent/dub.py

```diff
@@ -0,0 +1,221 @@
+                            "Suggest a detailed meal plan for the day, including breakfast, lunch, dinner, and snacks.",
+                            "Provide a brief explanation of why the plan is suited to the user's goals.",
+                            "Focus on clarity, coherence, and quality of the recommendations.",
+                        ]
+                    )
+
+                    fitness_agent = Agent(
+                        name="Fitness Expert",
+                        role="Provides personalized fitness recommendations",
+                        model=gemini_model,
+                        instructions=[
+                            "Provide exercises tailored to the user's goals.",
+                            "Include warm-up, main workout, and cool-down exercises.",
+                            "Explain the benefits of each recommended exercise.",
+                            "Ensure the plan is actionable and detailed.",
+                        ]
+                    )
+
+                    user_profile = f"""
+                    Age: {age}
+                    Weight: {weight}kg
+                    Height: {height}cm
+                    Sex: {sex}
+                    Activity Level: {activity_level}
+                    Dietary Preferences: {dietary_preferences}
+                    Fitness Goals: {fitness_goals}
+                    """
+
+                    # Generate plans
+                    dietary_plan_response = dietary_agent.run(user_profile)
+                    dietary_plan = {
+                        "why_this_plan_works": "High Protein, Healthy Fats, Moderate Carbohydrates, and Caloric Balance",
+                        "meal_plan": dietary_plan_response.content,
+                        "important_considerations": """
+                        - Hydration: Drink plenty of water throughout the day
+                        - Electrolytes: Monitor sodium, potassium, and magnesium levels
+                        - Fiber: Ensure adequate intake through vegetables and fruits
+                        - Listen to your body: Adjust portion sizes as needed
+                        """
+                    }
+
+                    fitness_plan_response = fitness_agent.run(user_profile)
+                    fitness_plan = {
+                        "goals": "Build strength, improve endurance, and maintain overall fitness",
+                        "routine": fitness_plan_response.content,
+                        "tips": """
+                        - Track your progress regularly
+                        - Allow proper rest between workouts
+                        - Focus on proper form
+                        - Stay consistent with your routine
+                        """
+                    }
+
+                    # Display plans
+                    display_dietary_plan(dietary_plan)
+                    display_fitness_plan(fitness_plan)
+
+                    # Q&A Section
```

Comment:
As soon as I click on ask question. It restarts the entire session. Can you please check this?

Reply:
The QA section

Reply:
yes, done!